src/haz3lcore/statics/Ctx.re

```diff
@@ -1,54 +1,60 @@
-open Sexplib.Std;
-
 [@deriving (show({with_path: false}), sexp, yojson)]
-type entry = {
-  id: Id.t,
-  typ: Typ.t,
+type t = {
+  term: TermCtx.t,
+  typ: TypCtx.t,
 };
 
 [@deriving (show({with_path: false}), sexp, yojson)]
-type t = VarMap.t_(entry);
-
-[@deriving (show({with_path: false}), sexp, yojson)]
-type co_item = {
-  id: Id.t,
-  mode: Typ.mode,
+type co = {
+  term: TermCtx.co,
+  typ: TypCtx.co,
 };
 
-[@deriving (show({with_path: false}), sexp, yojson)]
-type co_entry = list(co_item);
-
-[@deriving (show({with_path: false}), sexp, yojson)]
-type co = VarMap.t_(co_entry);
-
-let empty = VarMap.empty;
+let empty: t = {term: TermCtx.empty, typ: TypCtx.empty};
+let extend_term = (ctx: t, x: Token.t, term_entry): t => {
```

Comment:
why Token.t in some places and Id.t in others?

(In either case, don't we need to also have the unique ID in the domain of the map with the approach we have planned?)

Reply:
Our previous `Ctx.t` is a `VarMap.t_`, which takes `Token.t`s as keys. It's a map from `Token.t` to `{ id: Id.t, typ: Typ.t }`. This is why my rewrite tends to remain the same. I don't know if I'm missing anything... Afaik, `Ctx.t` is a map from `Token.t` to `Id.t` and it's `Typ.t` / `Kind.t`. If we want to lookup by id probably we should just look at `Statics.map` instead?